api/v1alpha1/connection_types.go

```diff
@@ -5,14 +5,23 @@
 
 package v1alpha1
 
-import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
 
 // Connection allows users to configure connection-level settings
 type Connection struct {
 	// ConnectionLimit defines limits related to connections
 	//
 	// +optional
 	ConnectionLimit *ConnectionLimit `json:"connectionLimit,omitempty"`
+	// BufferLimit provides configuration for the maximum buffer size for each incoming connection.
+	// For example, 128Mi, 500m, 2G, etc.
```

Comment:
looks like the type accepts a lot more such as `500m` is not relevant for this field.
can we limit it to specific types relevant for memory

Reply:
Do you mean, limit the use for only binary suffixes (`Ki | Mi | Gi | Ti | Pi | Ei`) and not decimal suffixed? (`m | "" | k | M | G | T | P | E`) 
If so, I can change the example and limit it using the kubebuilder schema validation.

Reply:
Maybe `*uint32` is a better option after all, validating the `*resource.Quantity` might be a bit complex (unnecessarily) :) 
Changed to `*uint32`.

Reply:
fwiw I did like the fact that the abstracted type was handling known quantities very well

Reply:
what do you think about runtime validation instead? users will get an error on the CTP but it won't prevent them from applying a resource with cpu specific suffixes

Reply:
sure, so the outlining the goal
* as a user, i'd like to specify mem in terms on absolute value (bytes), or known quantities (K, M, G, Ki, Mi, Gi) 

Options
* CEL validation for resource.Quanitity
* Runtime Validation of resource.Quanitity
* Creating our own type similar to what upstream did for Duration
https://github.com/kubernetes-sigs/gateway-api/blob/9183329ef9cff3f9222f001d2e112f325108c761/apis/v1/shared_types.go#L702

Reply:
Changed back to `Quantity` and added a cel validation. It will check that the suffix, if present, is compatible with the suffixes for memory: [units of memory in k8s](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory). 

One more thing, values without suffix, like `1500` (bytes) must be specified with `""` (otherwise there is an error). Is it something known to users as k8s behavior, or should I document it as well?

Reply:
can you elaborate on `""` , that feels like something that might annoy the user :) 
thoughts on the string approach with kubebuilder validations, and then typecasting into `resource.Quantity` so the user doesnt hit the gotcha ?

Reply:
Lior's suggestion solved the problem 👍